File: 1652-defuse-the-bomb/1652-defuse-the-bomb.py
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        n = len(code)
        if k == 0:
            return [0] * n
        
        extended_code = code + code
        result = [0] * n
        
        for i in range(n):
            if k > 0:
                result[i] = sum(extended_code[i + j + 1] for j in range(k))
            elif k < 0:
                result[i] = sum(extended_code[i + n + j] for j in range(k, 0))
        
        return result

File: 1652-defuse-the-bomb/1652-defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int n = code.size();
        if (k == 0) {
            return vector<int>(n, 0);
        }
        vector<int> extendedCode = code;
        extendedCode.insert(extendedCode.end(), code.begin(), code.end());
        vector<int> result(n, 0);
        if (k > 0) {
            for (int i = 0; i < n; i++) {
                int sum = 0;
                for (int j = 1; j <= k; j++) {
                    sum += extendedCode[i + j];
                }
                result[i] = sum;
            }
        } else {
            for (int i = 0; i < n; i++) {
                int sum = 0;
                for (int j = 1; j <= -k; j++) {
                    sum += extendedCode[i + n - j];
                }
                result[i] = sum;
            }
        }
        return result;
    }
};